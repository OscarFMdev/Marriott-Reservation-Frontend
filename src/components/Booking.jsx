import React from 'react';
import BookRoomForm from './forms/BookRoomForm';

const Booking = () => (
  <div>
    <BookRoomForm />
  </div>
);

export default Booking;
