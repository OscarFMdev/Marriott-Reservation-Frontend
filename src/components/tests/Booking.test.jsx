import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Booking from '../Booking';
import store from '../../redux/configureStore';

describe('It displays the Booking page title', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Router>
          <Booking />
        </Router>
      </Provider>,
    );
  });
  test('Booking title', () => {
    expect(screen.getByText('Book a Marriott Hotel Room')).toBeDefined();
  });
});

test('Booking page renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Booking />
      </Router>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
