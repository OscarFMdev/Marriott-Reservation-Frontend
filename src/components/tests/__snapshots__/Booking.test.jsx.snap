// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Booking page renders correctly 1`] = `
<div>
  <form
    className="_formContainer_6d2414"
    onSubmit={[Function]}
  >
    <div
      className="_hotel_6d2414"
    />
    <h1
      className="_header_6d2414"
    >
      Book a Marriott Hotel Room
    </h1>
    <p
      className="_details_6d2414"
    >
      Marriott is the largest hotel chain in the world. There are 3 room models in this Marriott Hotel: Single, Double and Suite. If you want to live the Marriott experience fill this form!
    </p>
    <p
      style={
        {
          "color": "red",
        }
      }
    />
    <div
      className="_inputsContainer_6d2414"
    >
      <div
        className="_fieldBook_6d2414"
      >
        <select
          id="hotelId"
          name="hotelId"
          onChange={[Function]}
          required={true}
          value=""
        >
          <option
            value=""
          >
            -- Select a Room --
          </option>
        </select>
      </div>
      <div
        className="_fieldBook_6d2414"
      >
        <label
          htmlFor="startDate"
        >
          <input
            id="startDate"
            name="startDate"
            onChange={[Function]}
            required={true}
            type="date"
            value=""
          />
        </label>
      </div>
      <div
        className="_last_6d2414"
      >
        <div
          className="_fieldBook_6d2414"
        >
          <label
            htmlFor="endDate"
          >
            <input
              id="endDate"
              name="endDate"
              onChange={[Function]}
              required={true}
              type="date"
              value=""
            />
          </label>
        </div>
        <button
          type="submit"
        >
          Book now
        </button>
      </div>
    </div>
  </form>
</div>
`;
